modules/dnn/src/layers/layers_common.simd.hpp

```diff
@@ -737,5 +737,148 @@ void fastGEMM( const float* aptr, size_t astep, const float* bptr,
+                vfloat32m2_t b0 = vle32_v_f32m2(bptr + k*bstep + n, vl);
+                vfloat32m2_t b1 = vle32_v_f32m2(bptr + k*bstep + n + 8, vl);
+                d00 = vfmacc_vv_f32m2(d00, a0, b0, vl);
+                d01 = vfmacc_vv_f32m2(d01, a0, b1, vl);
+                d10 = vfmacc_vv_f32m2(d10, a1, b0, vl);
+                d11 = vfmacc_vv_f32m2(d11, a1, b1, vl);
+                d20 = vfmacc_vv_f32m2(d20, a2, b0, vl);
+                d21 = vfmacc_vv_f32m2(d21, a2, b1, vl);
+                d30 = vfmacc_vv_f32m2(d30, a3, b0, vl);
+                d31 = vfmacc_vv_f32m2(d31, a3, b1, vl);
+            }
+            vse32_v_f32m2(cptr0 + n, d00, vl);
+            vse32_v_f32m2(cptr0 + n + 8, d01, vl);
+            vse32_v_f32m2(cptr1 + n, d10, vl);
+            vse32_v_f32m2(cptr1 + n + 8, d11, vl);
+            vse32_v_f32m2(cptr2 + n, d20, vl);
+            vse32_v_f32m2(cptr2 + n + 8, d21, vl);
+            vse32_v_f32m2(cptr3 + n, d30, vl);
+            vse32_v_f32m2(cptr3 + n + 8, d31, vl);
+        }
+    }
+
+    for( ; n < nb; n++ )
+    {
+        for( int m = 0; m < ma; m++ )
+        {
+            const float* aptr0 = aptr + astep*m;
+            float* cptr0 = cptr + cstep*m;
+            float d0 = 0.f;
+
+            for( int k = 0; k < na; k++ )
+                d0 += aptr0[k]*bptr[k*bstep + n];
+
+            cptr0[n] = d0;
+        }
+    }
+}
+
+void fastGEMM1T( const float* vec, const float* weights,
+                 size_t wstep, const float* bias,
+                 float* dst, int nvecs, int vecsize )
+{
+    int i = 0;
+    size_t vl = vsetvl_e32m2(128);
+    for( ; i <= nvecs - 8; i += 8 )
+    {
+        const float* wptr = weights + i*wstep;
+        vfloat32m2_t vs0 = vfmv_v_f_f32m2(0, vl), vs1 = vfmv_v_f_f32m2(0, vl),
+               vs2 = vfmv_v_f_f32m2(0, vl), vs3 = vfmv_v_f_f32m2(0, vl),
+               vs4 = vfmv_v_f_f32m2(0, vl), vs5 = vfmv_v_f_f32m2(0, vl),
+               vs6 = vfmv_v_f_f32m2(0, vl), vs7 = vfmv_v_f_f32m2(0, vl);
+
+        for( int k = 0; k < vecsize; k += 8, wptr += 8 )
+        {
+            vfloat32m2_t v = vle32_v_f32m2(vec + k, vl);
+
+            vs0 = vfmacc_vv_f32m2(vs0, vle32_v_f32m2(wptr, vl), v, vl);
+            vs1 = vfmacc_vv_f32m2(vs1, vle32_v_f32m2(wptr + wstep, vl), v, vl);
+            vs2 = vfmacc_vv_f32m2(vs2, vle32_v_f32m2(wptr + wstep*2, vl), v, vl);
+            vs3 = vfmacc_vv_f32m2(vs3, vle32_v_f32m2(wptr + wstep*3, vl), v, vl);
+            vs4 = vfmacc_vv_f32m2(vs4, vle32_v_f32m2(wptr + wstep*4, vl), v, vl);
+            vs5 = vfmacc_vv_f32m2(vs5, vle32_v_f32m2(wptr + wstep*5, vl), v, vl);
+            vs6 = vfmacc_vv_f32m2(vs6, vle32_v_f32m2(wptr + wstep*6, vl), v, vl);
+            vs7 = vfmacc_vv_f32m2(vs7, vle32_v_f32m2(wptr + wstep*7, vl), v, vl);
+        }
+
+        // Calculate the sum of each vector
+        vfloat32m1_t zero = vfmv_v_f_f32m1(0, vl);
+        vfloat32m1_t temp0 = vfredsum_vs_f32m2_f32m1(temp0, vs0, zero, vl);
```

Comment:
does this code compile? the initialization expression for temp0 references itself.

As a more general comment, probably the reduction can be optimized further. Is there a way to compute the reduced sum into a vector register and use one store operation instead of 8?